tm2py/controller.py

```diff
@@ -0,0 +1,144 @@
+"""RunController - model operation controller
+"""
+
+import argparse
+import itertools
+import os
+from typing import Union, List
+
+from tm2py.config import Configuration
+from tm2py.emme.manager import EmmeManager
+from tm2py.logger import Logger
+from tm2py.components.network.highway.highway_assign import HighwayAssignment
+from tm2py.components.network.highway.highway_network import PrepareNetwork
+from tm2py.components.network.highway.highway_maz import AssignMAZSPDemand, SkimMAZCosts
+
+
+# pylint: disable=too-many-instance-attributes
+
+
+class RunController:
+    """Main operational interface for model runs.
+
+    Provide one or more config files in TOML (*.toml) format, and a run directory.
+    If the run directory is not provided the root directory of the first config_file is used.
+    """
+
+    def __init__(self, config_file: Union[List[str], str] = None, run_dir: str = None):
+        if not isinstance(config_file, list):
+            config_file = [config_file]
+        if run_dir is None:
+            run_dir = os.path.abspath(os.path.dirname(config_file[0]))
+        self._run_dir = run_dir
+
+        self.config = Configuration(config_file)
+        self.logger = Logger(self)
+        self.top_sheet = None
+        self.trace = None
+
+        self.component_map = {
+            "prepare_network_highway": PrepareNetwork(self),
+            "highway": HighwayAssignment(self),
+            "highway_maz_assign": AssignMAZSPDemand(self),
+            "highway_maz_skim": SkimMAZCosts(self),
+        }
+        self.completed_components = []
+        self._emme_manager = None
+        self._iteration = None
+        self._component = None
+        self._queued_components = []
+        self._queue_components()
```

Comment:
My preferred way is to use the config.run.start_component with config.run.start_iteration in the scenario_config.toml. Going forward it would be pretty straight forward to make this less manual (enable auto-resume), but given the interest in running from command-line something file-based would be necessary and I generally prefer having one way to do things if at all possible.

Reply:
cc to @lmz as a food for thought for later. I will add this to an issue that we may or may not want to take up.